### PR TITLE
"No Results" flashing

### LINKED ISSUE #1923: onboarding flashes no results prior to loading first time
open app, tap get started, see zero state of follow artist screen "no results found" then the popular artist list flashes in.

### PR DESCRIPTION
**Pull request by ashfurrow**
Fixes #1923.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
  </tbody>
</table>

Time | Class | Function |
| --- | ----- | ----- |341.3ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
306.1ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Artsy/View_Controllers/Auction/AuctionViewController.swift#L199) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
238.8ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
112.7ms | [LiveAuctionLotViewController.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift#L47) | @objc override func viewDidLoad()
104.7ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
102.3ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)
100.5ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, salesPerson: LiveAuctionsSalesPersonType, sideMargin: String)
91.8ms | [Observable.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
78.0ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
74.4ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L40) | (closure)
67.2ms | [RefinementOptionsViewController+Private.swift](/artsy/eigen/tree/ashfurrow-no-results-flashing/Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift#L81) | func createTableView() -> UITableView


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: CHANGELOG.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ upcoming:
     dev:
       infrastructure:
         - Removed background fetch related code - alloy
+        - Fixed problem with "no results" label glashing during onboarding - ash
```
Comment (ashfurrow) [inline CHANGELOG.yml]: Should this go in `user_facing` instead?

Comment (alloy) [inline CHANGELOG.yml]: I was under the impression that beta-only changes (i.e. those that fix issues introduced during a previous beta) don’t need to be written up as for that an automatic git log would be used? Maybe I’m just making this all up… @orta?

Comment (orta) [inline CHANGELOG.yml]: ah, sorry, I did reply but it turned into a review. 

I think they're fine to go in, there's not an automatic git based log, it gets derived from this - the logs could be really verbose between betas

File: Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/5-_Onboarding_Personalization/AROnboardingPersonalizeTableViewController.m [hunk 2]
```diff
@@ -100,7 +104,8 @@ - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
 {
     if (self.searchResults.count == 0) {
-        self.noResultsLabel.hidden = NO;
+        BOOL resultsNotYetLoaded = (self.loadedInitialResults == NO);
+        self.noResultsLabel.hidden = resultsNotYetLoaded;
```
Comment (mennenia) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/5-_Onboarding_Personalization/AROnboardingPersonalizeTableViewController.m]: Would it make sense to assign in one go..?

Comment (ashfurrow) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/5-_Onboarding_Personalization/AROnboardingPersonalizeTableViewController.m]: I had that originally, but the boolean-to-English translation confused me (double-negative with `hidden`). I thought storing the value in a named variable would help readability. I can shorten this to one line, just let me know 👍

Comment (mennenia) [inline Artsy/View_Controllers/Login_and_Onboarding/Onboarding_stages/5-_Onboarding_Personalization/AROnboardingPersonalizeTableViewController.m]: nah it's no biggie - I just always get confused with bools so it took me a few tries to read this also.

**Comment by mennenia** [comment]
Thanks @ashfurrow !
